custom/enikshay/duplicate_ids.py

```diff
@@ -0,0 +1,96 @@
+from __future__ import absolute_import
+from __future__ import unicode_literals
+from collections import Counter
+import six
+
+from corehq.apps.users.models import CommCareUser
+from corehq.form_processor.backends.sql.dbaccessors import CaseAccessorSQL
+from corehq.form_processor.interfaces.dbaccessors import CaseAccessors
+from dimagi.utils.couch.database import iter_docs
+
+from .const import VOUCHER_ID, ENIKSHAY_ID
+
+
+def get_cases_with_duplicate_ids(domain, case_type, all_case_ids):
+    accessor = CaseAccessors(domain)
+    id_property = {'voucher': VOUCHER_ID, 'person': ENIKSHAY_ID}[case_type]
+    all_cases = [
+        {
+            'case_id': case.case_id,
+            'readable_id': case.get_case_property(id_property),
+            'opened_on': case.opened_on,
+        }
+        for case in accessor.iter_cases(all_case_ids)
+    ]
+    counts = Counter(case['readable_id'] for case in all_cases)
+    bad_cases = (case for case in all_cases if counts[case['readable_id']] > 1)
+    return sorted(bad_cases, key=lambda case: case['opened_on'], reverse=True)
+
+
+def get_bad_case_info(domain, case_type, full_debug_info=False):
+    accessor = CaseAccessors(domain)
+    case_ids = accessor.get_case_ids_in_domain(case_type)
+    bad_cases = get_cases_with_duplicate_ids(domain, case_type, case_ids)
+    add_debug_info_to_cases(bad_cases, full_debug_info)
+    context = {
+        'case_type': case_type,
+        'num_bad_cases': len(bad_cases),
+        'num_total_cases': len(case_ids),
+        'num_good_cases': len(case_ids) - len(bad_cases),
+        'bad_cases': bad_cases,
+    }
+    return context
+
+
+def add_debug_info_to_cases(bad_cases, full_debug_info):
+    _add_form_info_to_cases(bad_cases, full_debug_info)
+    _add_user_info_to_cases(bad_cases)
+
+
+def _add_form_info_to_cases(bad_cases, full_debug_info):
+    for case in bad_cases[:None if full_debug_info else 300]:
```

Comment:
What is 300?

Reply:
It's an arbitrary number of cases for which to include expensive debug info.  I moved this to the view and added a comment